webf/lib/src/dom/intersection_observer.dart

```diff
@@ -0,0 +1,159 @@
+    if (!element.addIntersectionObserver(this, _thresholds)) {
+      return;
+    }
+    _elementList.add(element);
+  }
+
+  void unobserve(Element element) {
+    // debugPrint('Dom.IntersectionObserver.unobserve');
+    _elementList.remove(element);
+    element.removeIntersectionObserver(this);
+  }
+
+  void disconnect() {
+    if (_elementList.isEmpty) return;
+    for (var element in _elementList) {
+      element!.removeIntersectionObserver(this);
+    }
+    _elementList.clear();
+  }
+
+  bool HasObservations() {
+    return _elementList.isNotEmpty;
+  }
+
+  void addEntry(DartIntersectionObserverEntry entry) {
+    _entries.add(entry);
+  }
+
+  List<DartIntersectionObserverEntry> takeRecords() {
+    List<DartIntersectionObserverEntry> entries = _entries.map((entry) => entry.copy()).toList();
+    _entries.clear();
+    return entries;
+  }
+
+  Future<void> deliver(WebFController controller) async {
+    if (pointer == null) return;
+
+    List<DartIntersectionObserverEntry> entries = takeRecords();
+    if (entries.isEmpty) {
+      return;
+    }
+    // debugPrint('Dom.IntersectionObserver.deliver size:${entries.length}');
+    Completer completer = Completer();
+
+    EvaluateOpItem? currentProfileOp;
+    if (enableWebFProfileTracking) {
+      currentProfileOp = WebFProfiler.instance.startTrackEvaluate('_dispatchEventToNative');
+    }
+
+    BindingObject bindingObject = controller.view.getBindingObject(pointer!);
+    // Call methods implements at C++ side.
+    DartInvokeBindingMethodsFromDart? invokeBindingMethodsFromDart =
+        pointer!.ref.invokeBindingMethodFromDart.asFunction();
+
+    Stopwatch? stopwatch;
+    if (enableWebFCommandLog) {
+      stopwatch = Stopwatch()..start();
+    }
+
+    // Allocate an chunk of memory for an list of NativeIntersectionObserverEntry
+    Pointer<NativeIntersectionObserverEntry> nativeEntries =
+        malloc.allocate(sizeOf<NativeIntersectionObserverEntry>() * entries.length);
+
+    // Write the native memory from dart objects.
+    for (int i = 0; i < entries.length; i++) {
+      (nativeEntries + i).ref.isIntersecting = entries[i].isIntersecting ? 1 : 0;
+      (nativeEntries + i).ref.intersectionRatio = entries[i].intersectionRatio;
+      (nativeEntries + i).ref.element = entries[i].element.pointer!;
+    }
+
+    List<dynamic> dispatchEntryArguments = [nativeEntries, entries.length];
+    Pointer<NativeValue> allocatedNativeArguments = makeNativeValueArguments(bindingObject, dispatchEntryArguments);
+
+    _IntersectionObserverDeliverContext context = _IntersectionObserverDeliverContext(
+        completer, stopwatch, allocatedNativeArguments, nativeEntries, controller, currentProfileOp);
+
+    Pointer<NativeFunction<NativeInvokeResultCallback>> resultCallback = Pointer.fromFunction(_handleDeliverResult);
+
+    Future.microtask(() {
+      invokeBindingMethodsFromDart(pointer!, currentProfileOp?.hashCode ?? 0, nullptr, dispatchEntryArguments.length,
```

Comment:
potential crash issue when the context or pointer was invalid.

reference:
https://github.com/openwebf/webf/blob/main/webf/lib/src/bridge/binding.dart#L110